osquery/config/config.cpp

```diff
@@ -270,53 +281,31 @@ void Config::getPerformanceStats(
   }
 }
 
+void Config::hashSource(const std::string& source, const std::string& content) {
+  std::vector<char> buffer;
```

Comment:
Can you just pass `buffer.c_str()` to the `hashFromBuffer` method? And avoid the content duplication?